CharlieBackend.Api/Controllers/LessonsController.cs

```diff
@@ -71,16 +71,16 @@ public async Task<ActionResult> AssignMentorToLesson(AssignMentorToLessonDto ids
         }
 
         /// <summary>
-        /// Gets list of all lessons
+        /// Get all lessons by date
         /// </summary>
-        /// <response code="200">Successful return of lessons list</response>
+        /// <response code="200">Successful return of lessons list by date</response>
         [SwaggerResponse(200, type: typeof(List<LessonDto>))]
         [Authorize(Roles = "Admin, Mentor, Secretary")]
-        [HttpGet]
-        public async Task<ActionResult<List<LessonDto>>> GetAllLessons()
+        [HttpGet("DateTime")]
```

Comment:
Please remove "DateTime". Also make the startDate and finishDate parameters optional.